spec/models/user_spec.rb

```diff
@@ -1,5 +1,22 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  it { should have_valid(:first_name).when('Anthony', 'Jane') }
+  it { should_not have_valid(:first_name).when(nil, '') }
+
+  it { should have_valid(:last_name).when('Watson', 'Jones') }
+  it { should_not have_valid(:last_name).when(nil, '') }
+
+  it { should have_valid(:email).when('anthony.jones@gmail.com',
+    'jane.watson@yahoo.com') }
+  it { should_not have_valid(:email).when(nil, 'jane', '.com', 'jane@net',
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.